src/navigationStore.js

```diff
@@ -281,16 +281,29 @@ class NavigationStore {
   reducer = null;
   router;
   _state;
-  _currentParams;
-  @observable currentScene = '';
-  @observable prevScene = '';
-  @observable currentParams;
-
+  // @observable currentScene = '';
+  // @observable prevScene = '';
+  // @observable currentParams;
+  currentScene = '';
+  prevScene = '';
+  currentParams;
+  engineDispatch;
+/*
   get state() {
     const scene = this.currentScene;// eslint-disable-line no-unused-vars
     const params = this.currentParams;// eslint-disable-line no-unused-vars
     return this._state;
   }
+*/
+  setupEngine(engine) {
+    engine.stateManager.setupHandlers(
+      this.nextState,
+      this.setState,
+      getActiveState
+    );
```

Comment:
I am not sure how I feel about this yet but I was stuck in a situation where this class needed to be able to call the engine (specifically the dispatch) and the engine needed to be able to call into this class. This was the simplest way I could think of for right now.

Reply:
Can't you just accept the engine constructor and inject self on instantiate? Good ol' DI? Then you have access to the child and the child has access to the parent.